lib/extras/dec/exr.cc

```diff
@@ -182,59 +237,129 @@ Status DecodeImageEXR(Span<const uint8_t> bytes, const ColorHints& color_hints,
   {
     JXL_ASSIGN_OR_RETURN(
         PackedFrame frame,
-        PackedFrame::Create(image_size.x, image_size.y, format));
+        PackedFrame::Create(imageWidth, imageHeight, format));
     ppf->frames.emplace_back(std::move(frame));
   }
-  const auto& frame = ppf->frames.back();
+  auto& frame = ppf->frames.back();
+
+  // Allocate extra channel images
+  uint32_t extraCount = 0;
+  size_t extraPixelBytes = 0;
+  std::set<const OpenEXR::Channel*> extraChannels;
+  for (OpenEXR::ChannelList::ConstIterator it = channels.begin(); it != channels.end(); ++it) {
+    const std::string name = it.name();
+    if (has_rgb && (name == chNameR || name == chNameG || name == chNameB))
+      continue;
+    if (has_alpha && name == chNameA)
+      continue;
+    if (name == chNameBase)
+      continue;
+
+    const bool fp16 = it.channel().type == OpenEXR::HALF;
+    ++extraCount;
+    extraPixelBytes += fp16 ? 2 : 4;
+    extraChannels.insert(&it.channel());
+
+    const JxlPixelFormat ec_format{1, fp16 ? JXL_TYPE_FLOAT16 : JXL_TYPE_FLOAT, JXL_NATIVE_ENDIAN, 0};
+    JXL_ASSIGN_OR_RETURN(
+        PackedImage ec,
+        PackedImage::Create(imageWidth, imageHeight, ec_format));
+    frame.extra_channels.emplace_back(std::move(ec));
+
+    PackedExtraChannel pec = {};
+    pec.ec_info.bits_per_sample = fp16 ? 16 : 32;
+    pec.ec_info.exponent_bits_per_sample = fp16 ? 5 : 8;
+    pec.ec_info.type = JXL_CHANNEL_OPTIONAL;
```

Comment:
This is fine, but consider adding a `// TODO`  here to map recognized extra channels to other predefined channel types (e.g. depth or thermal) if there are sufficiently commonly used naming conventions for those in EXR (I don't know).